include/nbl/ext/DepthPyramidGenerator/DepthPyramidGenerator.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_EXT_DEPTH_PYRAMID_GENERATOR_H_INCLUDED_
+#define _NBL_EXT_DEPTH_PYRAMID_GENERATOR_H_INCLUDED_
+
+#include "nabla.h"
+
+#include "nbl/video/IGPUImageView.h"
+#include "nbl/asset/format/EFormat.h"
+#include "../../../../Nabla/source/Nabla/COpenGLExtensionHandler.h"
+
+using namespace nbl;
+using namespace nbl::core;
+using namespace nbl::asset;
+using namespace nbl::video;
+
+namespace nbl
+{
+namespace ext
+{
+namespace DepthPyramidGenerator
+{
+
+// TODO: test: `E_MIPMAP_GENERATION_OPERATOR::BOTH`, `reundUpToPoTWithPadding = true`
+
+class DepthPyramidGenerator final
+{
+public:
+
+	enum class E_MIPMAP_GENERATION_OPERATOR
+	{
+		EMGO_MAX, 
+		EMGO_MIN,
+		EMGO_BOTH // min goes to r, max to g
+	};
+	
+	enum class E_WORK_GROUP_SIZE
+	{
+		EWGS_16x16x1 = 16u,
+		EWGS_32x32x1 = 32u
+	};
+
+	// TODO: more formats
+	enum class E_IMAGE_FORMAT
+	{
+		EIF_R16_FLOAT = EF_R16_SFLOAT,
+		EIF_R32_FLOAT = EF_R32_SFLOAT,
+		EIF_R16G16_FLOAT = EF_R16G16_SFLOAT,
+		EIF_R32G32_FLOAT = EF_R32G32_SFLOAT
+	};
```

Comment:
I dont see the point of having a special extra enum to narrow down an existing enum

all you need is just a validation function for `outputFormat`